shell/platform/fuchsia/flutter/platform_view.cc

```diff
@@ -122,33 +122,31 @@ PlatformView::PlatformView(
   });
 
   // Begin watching for pointer events.
-  if (is_flatland) {  // TODO(fxbug.dev/85125): make unconditional
-    pointer_delegate_->WatchLoop([weak = weak_factory_.GetWeakPtr()](
-                                     std::vector<flutter::PointerData> events) {
-      if (!weak) {
-        FML_LOG(WARNING) << "PlatformView use-after-free attempted. Ignoring.";
-        return;
-      }
+  pointer_delegate_->WatchLoop([weak = weak_factory_.GetWeakPtr()](
```

Comment:
Should this really be unconditional?  The MouseSource will still be invalid under gfx, and pointer_delegate_->WatchLoop is gonna try to use the MouseSource

Maybe we need to pass is_flatland into the pointer delegate?  Or, the pointer delegate could check if the MouseSource channel is valid before trying to use it.

Reply:
Yeah, I had started to thread the is_flatland bool into the PointerDelegate class, and it felt clumsy. I like your second idea - check MouseSource channel validity. Will give it a spin.

Reply:
Done. Logs remain clean with this conversion.